docs/en/sql-reference/statements/alter/user.md

```diff
@@ -62,3 +62,20 @@ Allows the user with `john` account to grant his privileges to the user with `ja
 ``` sql
 ALTER USER john GRANTEES jack;
 ```
+
+Adds new authentication methods to the user while keeping the existing ones
+
+``` sql
+ALTER USER user ADD IDENTIFIED WITH plaintext_password by '1' ADD IDENTIFIED WITH bcrypt_password by '2' ADD IDENTIFIED WITH plaintext_password by '3'
+```
+
+Reset authentication methods and adds the ones specified in the query (effect of leading IDENTIFIED without the ADD keyword)
+
+``` sql
+ALTER USER user IDENTIFIED WITH plaintext_password by '1' ADD IDENTIFIED WITH bcrypt_password by '2' ADD IDENTIFIED WITH plaintext_password by '3'
+```
+
+Reset authentication methods and keep the most recent added one
+``` sql
+ALTER USER user RESET AUTHENTICATION METHODS TO NEW
```

Comment:
This looks counterintuitive. In what scenario can this command be helpful?

Reply:
This comes from both Milovidov suggestion in https://github.com/ClickHouse/ClickHouse/issues/65136#issuecomment-2162096010 and https://www.percona.com/blog/using-mysql-8-dual-passwords/